modules/drivers/mongo/src/metabase/driver/mongo/execute.clj

```diff
@@ -161,30 +168,59 @@
                 (remaining-rows-thunk)))]
       (qp.reducible/reducible-rows row-thunk (qp.context/canceled-chan context)))))
 
-(defn- reduce-results [native-query query context ^Cursor cursor respond]
+(defn- interrupt-ex? [^MongoCommandException ex]
+  (and (= 11601 (.getErrorCode ex))
+       (= "Interrupted" (.getErrorCodeName ex))))
+
+(defn- reduce-results [native-query query context aggregate respond]
   (try
-    (let [first-row                        (when (.hasNext cursor)
-                                             (.next cursor))
-          {row-col-names       :row
-           unescaped-col-names :unescaped} (result-col-names native-query query (row-keys first-row))]
-      (log/tracef "Renaming columns in results %s -> %s" (pr-str row-col-names) (pr-str unescaped-col-names))
-      (respond (result-metadata unescaped-col-names)
-               (if-not first-row
-                 []
-                 (reducible-rows context cursor first-row (post-process-row row-col-names)))))
-    (finally
-      (.close cursor))))
+    (with-open [cursor ^MongoCursor (.cursor ^AggregateIterable aggregate)]
+      (let [first-row                        (when (.hasNext cursor)
+                                               (.next cursor))
+            {row-col-names       :row
+             unescaped-col-names :unescaped} (result-col-names native-query query (row-keys first-row))]
+        (log/tracef "Renaming columns in results %s -> %s" (pr-str row-col-names) (pr-str unescaped-col-names))
+        (respond (result-metadata unescaped-col-names)
+                 (if-not first-row
+                   []
+                   (reducible-rows context cursor first-row (post-process-row row-col-names))))))
+    (catch MongoCommandException ex
```

Comment:
Is it desirable to catch this exception? Is it desirable to log that information with a warning level? `interrupt-ex?` checks error code, so other exceptions are re-thrown.